### PR TITLE
Delete implicit dir integration test

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Adding tests for implicit directory deletion.
1. TestDeleteNonEmptyImplicitDir
2. TestDeleteNonEmptyImplicitSubDir
3. TestDeleteImplicitDirWithExplicitSubDir
4. TestDeleteImplicitDirWithExplicitSubDirInImplicitSubDir
5. TestDeleteImplicitDirInExplicitDir
6. TestDeleteExplicitDirWithImplicitSubDir

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Run all the tests for testbucket and mountedDirectory flags.(status:SUCCESS)
11. Unit tests - NA
12. Integration tests - Run all the tests for testbucket and mountedDirectory flags.(status:SUCCESS)

### CODE REVIEW

**Comment by raj-prince** [comment]
"TestDeleteExplicitDirWithImplicitSubDir" - Will this test not be part of normal directory deletion test?

File: tools/integration_tests/implicit_dir/delete_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+// testBucket/implicitDirectory/fileInImplicitDir1                               -- File
+// testBucket/implicitDirectory/implicitSubDirectory                             -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2          -- File
+func TestDeleteNonEmptyImplicitSubDir(t *testing.T) {
+	// Clean the mountedDirectory before running test.
+	setup.CleanMntDir()
+
+	implicit_and_explicit_dir_setup.CreateImplicitDirectory()
+
+	subDirPath := path.Join(setup.MntDir(), implicit_and_explicit_dir_setup.ImplicitDirectory, implicit_and_explicit_dir_setup.ImplicitSubDirectory)
```
Comment (raj-prince) [inline tools/integration_tests/implicit_dir/delete_test.go:54]: How are we creating implicit subDir?

File: tools/integration_tests/implicit_dir/delete_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+// Directory Structure
+// testBucket/implicitDirectory                                                                                         -- Dir
+// testBucket/implicitDirectory/fileInImplicitDir1                                                                      -- File
+// testBucket/implicitDirectory/implicitSubDirectory                                                                    -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/fileInImplicitDir2                                                 -- File
+// testBucket/implicitDirectory/implicitSubDirectory/explicitDirInImplicitDir                                           -- Dir
+// testBucket/implicitDirectory/implicitSubDirectory/explicitDirInImplicitDir/fileInExplicitDirInImplicitDir            -- File
+func TestDeleteImplicitDirWithExplicitSubDirInImplicitSubDir(t *testing.T) {
```
Comment (raj-prince) [inline tools/integration_tests/implicit_dir/delete_test.go:88]: Please rename it to - TestDeleteImplicitDirWithImplicitSubDirContainingExplicitDir.

File: tools/integration_tests/implicit_dir/delete_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+// Directory Structure
+// testBucket/explicitDirectory                                                                   -- Dir
+// testBucket/explictFile                                                                         -- File
+// testBucket/explicitDirectory/fileInExplicitDir1                                                -- File
+// testBucket/explicitDirectory/fileInExplicitDir2                                                -- File
+// testBucket/explicitDirectory/implicitDirectory                                                 -- Dir
+// testBucket/explicitDirectory/implicitDirectory/fileInImplicitDir1                              -- File
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory                            -- Dir
+// testBucket/explicitDirectory/implicitDirectory/implicitSubDirectory/fileInImplicitDir2         -- File
+func TestDeleteExplicitDirWithImplicitSubDir(t *testing.T) {
```
Comment (raj-prince) [inline tools/integration_tests/implicit_dir/delete_test.go:131]: What's the difference b/w this test and the above test (TestDeleteImplicitDirInExplicitDir)?

**Review by raj-prince — COMMENTED** [review]
Added few comments please check.

File: tools/integration_tests/implicit_dir/delete_test.go (hunk 3 above)
Comment (Tulsishah) [reply tools/integration_tests/implicit_dir/delete_test.go:131]: Above test is removing implicit directory from explicit directory
e.g
imp1
imp1/expsub
imp1/expsub/impsub2

remove(impsub2)

This test is removing explicit directory

e.g
imp1
imp1/expsub
imp1/expsub/impsub2

remove(expsub)

**Comment by Tulsishah** [comment]
> "6. TestDeleteExplicitDirWithImplicitSubDir" - Will this test not be part of normal directory deletion test?

In normal tests Explicit dir does not contains implicit dir. That's why I added here. For adding implicit structure we can do it from script only(gsutil)

File: tools/integration_tests/implicit_dir/delete_test.go (hunk 1 above)
Comment (Tulsishah) [reply tools/integration_tests/implicit_dir/delete_test.go:54]: [This](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/util/setup/implicit_and_explicit_dir_setup/testdata/create_objects.sh) script is creating whole structure.

Comment (raj-prince) [reply tools/integration_tests/implicit_dir/delete_test.go:54]: Okay, got it. Then this name (CreateImplicitDirectory) is misleading, feels like one empty implicit directory. Instead, we should rename it to CreateImplicitDirectoryStructure. 

Similarly for CreateExplicitDirectoryStructure.

File: tools/integration_tests/implicit_dir/delete_test.go (hunk 3 above)
Comment (raj-prince) [reply tools/integration_tests/implicit_dir/delete_test.go:131]: Okay. Directory structure is same, and we are deleting different object.

We should rename it to -> TestDeleteExplicitDirContainingImplicitSubDir

**Pull request merged**